core/controllers/library_test.py

```diff
@@ -187,6 +188,62 @@ def test_library_handler_for_created_explorations(self):
         }, response_dict['activity_list'][0])
 
 
+class LibraryGroupPageTest(test_utils.GenericTestBase):
+
+    def test_library_group_pages(self):
+        """Test access to the top rated and recently published pages."""
+        response = self.testapp.get(feconf.LIBRARY_TOP_RATED_URL)
+        self.assertEqual(response.status_int, 200)
+
+        response = self.testapp.get(feconf.LIBRARY_RECENTLY_PUBLISHED_URL)
+        self.assertEqual(response.status_int, 200)
+
+    def test_handler_for_library_group_pages(self):
+        """Test library handler for top rated and recently published pages."""
+        response_dict = self.get_json(
+            feconf.LIBRARY_GROUP_DATA_URL,
+            {'group_name': feconf.LIBRARY_GROUP_RECENTLY_PUBLISHED})
+        self.assertDictContainsSubset({
+            'is_admin': False,
+            'is_moderator': False,
+            'is_super_admin': False,
+            'activity_list': [],
+            'preferred_language_codes': ['en'],
+            'profile_picture_data_url': None,
+        }, response_dict)
+
+        # Load a public demo exploration.
+        exp_services.load_demo('0')
+
+        response_dict = self.get_json(
+            feconf.LIBRARY_GROUP_DATA_URL,
+            {'group_name': feconf.LIBRARY_GROUP_RECENTLY_PUBLISHED})
+        self.assertDictContainsSubset({
```

Comment:
Would having two tests in this order suffice:
```
        self.assertDictContainsSubset({
            'header_i18n_id': 'I18N_LIBRARY_GROUPS_RECENTLY_PUBLISHED',
            'preferred_language_codes': ['en'],
        }, response_dict)
        self.assertDictContainsSubset({
            'id': '0',
            'category': 'Welcome',
            'title': 'Welcome to Oppia!',
            'language_code': 'en',
            'objective': 'become familiar with Oppia\'s capabilities',
            'status': rights_manager.ACTIVITY_STATUS_PUBLIC,
        }, response_dict['activity_list'][0])
```
Testing activity_list along with header_i18n_id and preferred_language_codes is difficult as I only want to check if activity_list has a subset of the keys. activity_list has many other keys such as `'created_on_msec': 1475403322644.269` which we do not want to check.